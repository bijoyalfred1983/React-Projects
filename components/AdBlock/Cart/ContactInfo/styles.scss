
@import '../../../theme/variables';

.contactInfoContainer {
  width: 100%;
  padding: 59px 0;
  
  .questionTitle {
    font-family: $font-az-medium;
    font-size: 16px;
    font-weight: 500;
    font-style: normal;
    font-stretch: condensed;
    line-height: 1.2;
    letter-spacing: normal;
    text-transform: uppercase;
  }

  .contactInfo {
    font-family: $font-az-app;
    font-size: 14px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: normal;
    letter-spacing: 0.1px;
    margin-top: 5px;
  }

  .contactLinkText {
    text-decoration: underline;
    color: $black-bg-color;
  }
}

@media #{$extra-large-and-down} {
  .contactInfoContainer {
    text-align: center;
    padding: 40px 0;
    background-color: $color-white-main;
    margin-top: 24px;
	
    .questionTitle {
      width: 288px;
      font-size: 20px;   
      margin: auto;   
      color: $grey-ultra-dark;
    }

    .contactInfo {
      width: 288px;
      margin: auto;   
      line-height: 1.43;
      color: $grey-ultra-dark;
    }

    .contactLinkText {
      text-decoration: none;
      color: $grey-ultra-dark;
    }
  }
}

@media #{$medium-only} {
  .contactInfoContainer {
    background-color: $az-grey-light;	
    
	
    .questionTitle {
      width: auto;
      padding-left: 20px;
      text-align: left;	  
    }
	
    .contactInfo {
      width: auto;
      padding-left: 20px;
      text-align: left;
    }
  }
}
