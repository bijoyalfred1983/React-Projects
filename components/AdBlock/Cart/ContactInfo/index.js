/* @flow */

import React from 'react';
import styles from './styles.scss';

type Props = {
  staticLabelData: Object
};

export default (props: Props) => (
  <div className={styles.contactInfoContainer}>
    <div className={styles.questionTitle}>
      {props.staticLabelData.label_checkout_cart_QuestionsAllCaps
        ? props.staticLabelData.label_checkout_cart_QuestionsAllCaps
        : 'label_checkout_cart_QuestionsAllCaps'}
    </div>
    <div className={styles.contactInfo}>
      {props.staticLabelData.label_checkout_footer_ContactUsAt8002886966
        ? props.staticLabelData.label_checkout_footer_ContactUsAt8002886966
        : 'label_checkout_footer_ContactUsAt8002886966'}
      {' or '}
      <a
        className={styles.contactLinkText}
        href="mailto:customer.service@autozone.com"
      >
        {props.staticLabelData.label_checkout_footer_CustomerServiceAutozoneCom
          ? props.staticLabelData
              .label_checkout_footer_CustomerServiceAutozoneCom
          : 'label_checkout_footer_CustomerServiceAutozoneCom'}
      </a>{' '}
    </div>
  </div>
);
