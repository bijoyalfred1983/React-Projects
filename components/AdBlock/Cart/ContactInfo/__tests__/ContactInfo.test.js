import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import ContactInfo from '../';

 let PROPS = {};
describe('<ContactInfo />', () => {
	PROPS = {
		staticLabelData: {}
	}
  test('renders correctly ContactInfo', () => {
    const wrapper = mount(<ContactInfo {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

    test('renders correctly ContactInfo with static label data', () => {
   	PROPS.staticLabelData = {"label_checkout_cart_QuestionsAllCaps":"a",
   	"label_checkout_footer_ContactUsAt8002886966":"",
   	"label_checkout_footer_CustomerServiceAutozoneCom":"a",
   	"label_checkout_footer_Or4400SummerAvenueMemphisTn38122":"a"}
    const wrapper = mount(<ContactInfo {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
