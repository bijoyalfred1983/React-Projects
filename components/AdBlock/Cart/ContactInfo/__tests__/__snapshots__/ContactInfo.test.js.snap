// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactInfo /> renders correctly ContactInfo 1`] = `
<Component>
  <div
    className="contactInfoContainer"
  >
    <div
      className="questionTitle"
    >
      Questions?
    </div>
    <div
      className="contactInfo"
    >
      Contact us at (800) 288-6966,
       
      <a
        className="contactLinkText"
        href="mailto:customer.service@autozone.com"
      >
        customer.service@autozone.com
      </a>
       
      or 4400 Summer Avenue, Memphis TN 38122.
    </div>
  </div>
</Component>
`;

exports[`<ContactInfo /> renders correctly ContactInfo with static label data 1`] = `
<Component
  staticLabelData={
    Object {
      "label_checkout_cart_QuestionsAllCaps": "a",
      "label_checkout_footer_ContactUsAt8002886966": "",
      "label_checkout_footer_CustomerServiceAutozoneCom": "a",
      "label_checkout_footer_Or4400SummerAvenueMemphisTn38122": "a",
    }
  }
>
  <div
    className="contactInfoContainer"
  >
    <div
      className="questionTitle"
    >
      a
    </div>
    <div
      className="contactInfo"
    >
      label_checkout_footer_ContactUsAt8002886966
       
      <a
        className="contactLinkText"
        href="mailto:customer.service@autozone.com"
      >
        a
      </a>
       
      a
    </div>
  </div>
</Component>
`;
